File: app/src/main/java/com/gameondigital/gameonapp/Login/presenter/LoginPresenterContract.java
package com.gameondigital.gameonapp.Login.presenter;

import com.gameondigital.gameonapp.Login.ui.LoginActivity;

public interface LoginPresenterContract {
    void setFonts();

    void getCurrentUser();

    void setPersistenceFirebaseDatabase();

    void initFirebaseAuth();

    void showLoginView();

    void callHome();

    void validateLogin();

    void login(LoginActivity loginActivity, String email, String password);

    void callRegister(Class<?> cls);

    void callRememberPassword(Class<?> cls);

    void showErrorLogin(String errorCode);

    void dismissLoading();
}

File: app/src/main/java/com/gameondigital/gameonapp/Login/presenter/LoginPresenterContract.kt
package com.gameondigital.gameonapp.Login.presenter

import android.widget.EditText

interface LoginPresenterContract {
    fun initComponents()
    fun validatedLogin(email: String, password: String)
    fun goToHome()
    fun setValidateLoginError(edtText: EditText, error: String)
    fun showErrorLogin(msgError: String)
    fun showLoginView()
}